integration-tests/bats/profile.bats

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env bats
+load $BATS_TEST_DIRNAME/helper/common.bash
+load $BATS_TEST_DIRNAME/helper/query-server-common.bash
+
+make_repo() {
+  mkdir "$1"
+  cd "$1"
+  dolt init
+  dolt sql -q "create table $1_tbl (id int)"
+  dolt sql <<SQL
+CREATE TABLE table1 (pk int PRIMARY KEY);
+CREATE TABLE table2 (pk int PRIMARY KEY);
+INSERT INTO dolt_ignore VALUES ('generated_*', 1);
+SQL
+  dolt add -A && dolt commit -m "tables table1, table2"
+  dolt sql <<SQL
+INSERT INTO  table1 VALUES (1),(2),(3);
+INSERT INTO  table2 VALUES (1),(2),(3);
+CREATE TABLE table3 (pk int PRIMARY KEY);
+CREATE TABLE generated_foo (pk int PRIMARY KEY);
+SQL
+  dolt add table1
+  # Note that we leave the table in a dirty state, which is useful to several tests, and harmless to others. For
+  # some, you need to ensure the repo is clean, and you should run `dolt reset --hard` at the beginning of the test.
+  cd ..
+}
+
+setup() {
+    setup_no_dolt_init
+    make_repo defaultDB
+    make_repo altDB
+
+    dolt config --global --add profile '{"defaultTest": {"use-db":"defaultDB"}}'
```

Comment:
This ends up being a string within a JSON object which is a serialized json object, right? That's kinda gross.

I tested putting an object into config_global.json, and `dolt config` panics attempting to marshal that data into a string. Effective makes it impossible to run dolt, so that's a wrinkle in this plan.